### ISSUE TITLE
feature: Requesting support for node16/nodenext module resolution

### ISSUE THREAD

**Issue by kf6kjg** [MichalLytek/type-graphql#1441]
EDIT: Fixed in #1400 - thanks @carlocorradini!

### Description
When using this library in a project that has the TSConfig setting [`moduleResolution` set to `node16`](https://www.typescriptlang.org/docs/handbook/module-resolution.html) the TS compiler throws errors on every `import` and `export ... from` call.

Example:

```
../../node_modules/type-graphql/dist/index.d.ts(1,15): error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.
```

### Proposed solution

1. Change the `tsconfig.json#/compilerOptions/moduleResolution` setting to `node16`.
2. Bulk alter the codebase to correct the `import` and `export` paths to have the full path to the target file with the extension.
3. Manually fix the outliers in the code: converting implicit `index.js` into explicit being one such.

This should have no deleterious effect since, as far as I am aware, no version of TS or Node cared about having the filename and extension until ES6 came along.

**Comment by kf6kjg** [comment]
I'd read through that PR to see if it had this before posting, but not checked the code. So am I correct you are resolving this via using the `@/` paths option so that you can use absolute paths instead of relative ones, thus bypassing the error without switching to the `node16` module resolution strategy?

**Comment by carlocorradini** [comment]
Nope.
The code supports both CJS and ESM.
Moreover, all absolute paths are transformed to relative.
You can check it by executing:
```sh
npm run build
```

**Comment by carlocorradini** [comment]
If something is not clear let me know and I'll try to explain it better 🥳

**Comment by carlocorradini** [comment]
We can have `.js` in declarations too.
Wait 5 minutes. I'll push a fix and then you can try ok

**Comment by carlocorradini** [comment]
@kf6kjg Try now

**Comment by kf6kjg** [comment]
That looks good!

**Comment by kf6kjg** [comment]
Next up is looking for / opening a ticket for moving away from the types being in a separate folder to solve the other problem I'm having... :P

**Comment by carlocorradini** [comment]
What are the other problems?

**Comment by kf6kjg** [comment]
I've created https://github.com/MichalLytek/type-graphql/issues/1442 to track that problem space as it's orthogonal to this ticket. :)

**Comment by MichalLytek** [comment]
I think this can be closed by #1400 🔒 

@carlocorradini Can you share your example repo with ESM usage?
